framework/src/problems/FEProblemBase.C

```diff
@@ -4699,7 +4699,7 @@ FEProblemBase::execMultiApps(ExecFlagType type, bool auto_advance)
   // Execute MultiApps
   if (multi_apps.size())
   {
-    TIME_SECTION("execMultiApps", 1, "Executing MultiApps", false);
+    TIME_SECTION("execMultiApps", 1, "Executing MultiApps on " + type.name(), false);
```

Comment:
We simply cannot do this with the static registry. The name must be static.

Reply:
fixed

Reply:
And for context in case someone comes across this later - because the registry is static, you only get one ID per instantiation in code. Which means everything that runs from that point will take the first statically registered name. Later names will do nothing